src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -375,125 +377,126 @@ public async Task<JObject> Resolve(ElementAccessExpressionSyntax elementAccess,
+            {
+                throw new Exception($"Unable to evaluate method '{methodName}'", ex);
+            }
+        }
 
-                if (rootObject != null)
+        private static readonly string[] primitiveTypes = new string[] { "string", "number", "boolean", "symbol" };
+
+        public async Task<JObject> Resolve(InvocationExpressionSyntax method, Dictionary<string, JObject> memberAccessValues, CancellationToken token)
+        {
+            (JObject rootObject, string methodName) = await ResolveInvokationInfo(method, token);
+            if (rootObject == null)
+                throw new ReturnAsErrorException($"Failed to resolve root object for {method}", "ReferenceError");
+
+            // primitives don't have objectId
+            if (!DotnetObjectId.TryParse(rootObject["objectId"]?.Value<string>(), out DotnetObjectId objectId) &&
+                 primitiveTypes.Contains(rootObject["type"]?.Value<string>()))
+                return null;
+
+            if (method.ArgumentList == null)
+                throw new InternalErrorException($"Failed to resolve method call for {method}, list of arguments is null.");
+
+            bool isExtensionMethod = false;
+            try
+            {
+                var typeIds = await context.SdbAgent.GetTypeIdsForObject(objectId.Value, true, token);
+                int methodId = await context.SdbAgent.GetMethodIdByName(typeIds[0], methodName, token);
+                var className = await context.SdbAgent.GetTypeNameOriginal(typeIds[0], token);
+                if (methodId == 0) //try to search on System.Linq.Enumerable
+                    methodId = await FindMethodIdOnLinqEnumerable(typeIds, methodName);
+
+                if (methodId == 0)
+                {
+                    var typeName = await context.SdbAgent.GetTypeName(typeIds[0], token);
+                    throw new ReturnAsErrorException($"Method '{methodName}' not found in type '{typeName}'", "ReferenceError");
+                }
+                using var commandParamsObjWriter = new MonoBinaryWriter();
+                if (!isExtensionMethod)
                 {
-                    if (!DotnetObjectId.TryParse(rootObject?["objectId"]?.Value<string>(), out DotnetObjectId objectId))
-                        throw new ExpressionEvaluationFailedException($"Cannot invoke method '{methodName}' on invalid object id: {rootObject}");
+                    // instance method
+                    commandParamsObjWriter.WriteObj(objectId, context.SdbAgent);
+                }
 
-                    List<int> typeIds;
-                    if (objectId.IsValueType)
+                int passedArgsCnt = method.ArgumentList.Arguments.Count;
+                int methodParamsCnt = passedArgsCnt;
+                ParameterInfo[] methodParamsInfo = null;
+                var methodInfo = await context.SdbAgent.GetMethodInfo(methodId, token);
+                if (methodInfo != null) //FIXME: #65670
```

Comment:
`65670` is about optional params. It doesn't seem related here.

Reply:
It's already fixed, I should have removed this comment. The Issue number is correct, we needed optional params support for external functions to run tests on primitive for: str.Split(char, partsCnt, optional_enum: splitType). That's why this is here. This PR turned out to solve this case, so I'm removing this comment.